project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/organisations/CachedOrganisationRepository.java

```diff
@@ -81,27 +103,45 @@ private Optional<Organisation> lookupROR(String iri) {
     return extractRorId(iri).map(this::findOrganisationInROR).or(Optional::empty);
   }
 
+  private Optional<String> apiKey() {
+    return Optional.ofNullable(apiClientId);
+  }
+
   private Organisation findOrganisationInROR(String rorId) {
     try {
       HttpClient client = HttpClient.newBuilder().version(Version.HTTP_2)
           .followRedirects(Redirect.NORMAL).connectTimeout(
               Duration.ofSeconds(10)).build();
-      HttpRequest rorQuery = HttpRequest.newBuilder().uri(URI.create(ROR_API_URL.formatted(rorId)))
-          .header("Content-Type", "application/json").GET().build();
+
+      var queryBuilder = HttpRequest.newBuilder().uri(URI.create(ROR_API_URL.formatted(rorId)))
```

Comment:
This will lead to runtime exceptions, since the ROR_API_URL can be an invalid URL. I suggest to validate the URI and make it configurable-

Reply:
URI.create will fail for invalid URLs. I do not get your point here?